### PR TITLE
docs(devel-guide): Add provider section and remove audit_info section

### PR DESCRIPTION
**Pull request by puchy22**
### Context

The audit_info is no longer used in v4, change it for service section.

### Description

Remove the old `audit_info.md`, add new `provider.md`, changes in the nav bar and fix links from services adding the new k8s service.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/developer-guide/provider.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+
+# Create a new Provider for Prowler
+
+Here you can find how to create a new Provider in Prowler to give support for making all security checks needed and make your cloud safer!
+
+## Introduction
+
+Providers are the foundation on which Prowler is built, a simple definition for a cloud provider could be "third-party company that offers a platform where any IT resource you need is available at any time upon request". The most well-known cloud providers are Amazon Web Service, Azure from Microsoft and Google Cloud which are already supported by Prowler.
+
+To create a new provider that is not supported now by Prowler and add your security checks must create a new folder where store all related with it (services, checks, etc.). It must be store in route `prowler/providers/<new_provider_name>/`.
```
Comment (pedrooot) [inline docs/developer-guide/provider.md]:
```suggestion
To create a new provider that is not supported now by Prowler and add your security checks must create a new folder where store all related with it (services, checks, etc.). It must be stored in `prowler/providers/<new_provider_name>/`.
```

File: docs/developer-guide/provider.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+- A `lib` folder: to store all extra functions.
+- A `services` folder: to store all [services](./services.md) to audit.
+- An empty `__init__.py`: to make Python treat this service folder as a package.
+- A `<new_provider_name>_provider.py`, containing all the provider's logic necessary to get authenticated in the provider, configurations and extra data useful for final report.
+- A `models.py`, containing all the models necessary for the new provider.
+
+## Provider
+
+The Prowler's provider structure is the following and the way to use it is just in the generic service of each provider passing as a parameter of constructor to initialize all necessary session values.
```
Comment (pedrooot) [inline docs/developer-guide/provider.md]:
```suggestion
The structure for Prowler's providers is set up in such a way that they can be utilized through a generic service specific to each provider. This is achieved by passing the required parameters to the constructor, which in turn initializes all the necessary session values.
```

File: docs/developer-guide/provider.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+### Base Class
+
+All the providers in Prowler inherits from the same [base class](https://github.com/prowler-cloud/prowler/blob/master/prowler/providers/common/provider.py). It is an [abstract base class](https://docs.python.org/3/library/abc.html) that defines the interface for all provider classes in the auditing system.
+
+### Provider Class
+
+Due to the complexity and differencies of each provider use the rest of the providers as a template for the implementation.
```
Comment (pedrooot) [inline docs/developer-guide/provider.md]:
```suggestion
Due to the complexity and differences of each provider use the rest of the providers as a template for the implementation.
```

**Review by pedrooot — CHANGES_REQUESTED** [review]
Nice job! @puchy22 Please, review my comments.

File: docs/developer-guide/provider.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+
+# Create a new Provider for Prowler
+
+Here you can find how to create a new Provider in Prowler to give support for making all security checks needed and make your cloud safer!
+
+## Introduction
+
+Providers are the foundation on which Prowler is built, a simple definition for a cloud provider could be "third-party company that offers a platform where any IT resource you need is available at any time upon request". The most well-known cloud providers are Amazon Web Service, Azure from Microsoft and Google Cloud which are already supported by Prowler.
```
Comment (jfagoagas) [inline docs/developer-guide/provider.md]:
```suggestion
Providers are the foundation on which Prowler is built, a simple definition for a cloud provider could be "third-party company that offers a platform where any IT resource you need is available at any time upon request". The most well-known cloud providers are Amazon Web Services, Azure from Microsoft and Google Cloud which are already supported by Prowler.
```

File: docs/developer-guide/provider.md (hunk 1 above)
Comment (jfagoagas) [inline docs/developer-guide/provider.md]:
```suggestion
To create a new provider that is not supported now by Prowler and add your security checks you must create a new folder to store all the related files within it (services, checks, etc.). It must be store in route `prowler/providers/<new_provider_name>/`.
```

File: docs/developer-guide/provider.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+- A `<new_provider_name>_provider.py`, containing all the provider's logic necessary to get authenticated in the provider, configurations and extra data useful for final report.
+- A `models.py`, containing all the models necessary for the new provider.
+
+## Provider
+
+The Prowler's provider structure is the following and the way to use it is just in the generic service of each provider passing as a parameter of constructor to initialize all necessary session values.
+
+### Base Class
+
+All the providers in Prowler inherits from the same [base class](https://github.com/prowler-cloud/prowler/blob/master/prowler/providers/common/provider.py). It is an [abstract base class](https://docs.python.org/3/library/abc.html) that defines the interface for all provider classes in the auditing system.
```
Comment (jfagoagas) [inline docs/developer-guide/provider.md]:
```suggestion
All the providers in Prowler inherits from the same [base class](https://github.com/prowler-cloud/prowler/blob/master/prowler/providers/common/provider.py). It is an [abstract base class](https://docs.python.org/3/library/abc.html) that defines the interface for all provider classes.
```

Comment (jfagoagas) [reply docs/developer-guide/provider.md]: Here I'd explain the characteristics of the provider base class and explain what methods are enforced, also you can include a generic provider in pseudo-code to explain that first we need to gather arguments, then setup authentication, load the config, etc.

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Great work @puchy22 !! Please review my comments when you get a chance. Thanks!

File: docs/developer-guide/provider.md [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+### Provider Class
+
+Due to the complexity and differences of each provider use the rest of the providers as a template for the implementation.
+
+- [AWS](https://github.com/prowler-cloud/prowler/blob/master/prowler/providers/aws/aws_provider.py)
+- [GCP](https://github.com/prowler-cloud/prowler/blob/master/prowler/providers/gcp/gcp_provider.py)
+- [Azure](https://github.com/prowler-cloud/prowler/blob/master/prowler/providers/azure/azure_provider.py)
+- [Kubernetes](https://github.com/prowler-cloud/prowler/blob/master/prowler/providers/kubernetes/kubernetes_provider.py)
+
+To facilitate understanding here is a pseudocode of how could be the most basic provider could be with examples.
```
Comment (jfagoagas) [inline docs/developer-guide/provider.md]:
```suggestion
To facilitate understanding here is a pseudocode of how the most basic provider could be with examples.
```

**Review by jfagoagas — APPROVED** [review]
Awesome @puchy22, this is a great improvement in our Developer Guide, thanks for taking care of this 👏

**Review by pedrooot — APPROVED** [review]
🔝 🔝 🔝  Nice job @puchy22

**Pull request merged**